### PR TITLE
Pgauns/task/callcomposite 1 n locator

### PR DESCRIPTION
**Pull request by pratimag1289**
## Purpose
Open API for contoso to start a 1:N call. Changes includes
1. Created StartCallOptions with participants list.
2. Send StartCallOptions in RemoteOptions
3. Test API via demo

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Pull Request Checklist

- [ ] Public API changes
- [ ] Verified for cross-platform
- [ ] Synced with iOS, Web team
- [ ] Internal review completed
- [ ] UI Changes
  - [ ] Screen captures included
  - [ ] Tested for Light/Dark mode
  - [ ] Tested for screen rotation
  - [ ] Tested on Tablet and small screen device (5")
  - [ ] Include localization changes
- [ ] Tests
  - [ ] Memory leak analysis performed
  - [ ] Tested on API 21, 26 and latest 
  - [ ] Tested for background mode
  - [ ] Unit Tests Included
  - [ ] UI Automated Tests Included

## How to Test

* Open...
* Click on...

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/service/sdk/CallingSDKWrapper.kt [hunk 1]
```diff
@@ -186,12 +191,24 @@ internal class CallingSDKWrapper(
...
+                    callConfig.participants?.let {
+                        startCall(agent, audioOptions, videoOptions, it)
+                    }
                 }.exceptionally { error ->
                     onJoinCallFailed(startCallCompletableFuture, error)
                 }
             } else {
-                joinCall(agent, audioOptions, videoOptions, callLocator)
+                callLocator?.let {
+                    joinCall(agent, audioOptions, videoOptions, callLocator)
```
Comment (pavelprystinka) [inline azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/service/sdk/CallingSDKWrapper.kt:206]: Let's use explicit joinCall(agent, audioOptions, NULL, callLocator) as videoOptions is always null here

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/service/sdk/CallingSDKWrapper.kt [hunk 2]
```diff
@@ -172,9 +174,12 @@ internal class CallingSDKWrapper(
...
             audioOptions.isMuted = (audioState.operation != AudioOperationalStatus.ON)
-            val callLocator: JoinMeetingLocator = when (callConfig.callType) {
+            val callLocator: JoinMeetingLocator? = when (callConfig.callType) {
                 CallType.GROUP_CALL -> GroupCallLocator(callConfig.groupId)
                 CallType.TEAMS_MEETING -> TeamsMeetingLinkLocator(callConfig.meetingLink)
+                else -> {
+                    null
+                }
             }
             var videoOptions: VideoOptions? = null
```
Comment (pavelprystinka) [inline azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/service/sdk/CallingSDKWrapper.kt:184]: Please move var videoOptions: VideoOptions? = null inside of getLocalVideoStream().whenComplete

**Review by pavelprystinka — CHANGES_REQUESTED** [review]

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/configuration/CallConfiguration.kt [hunk 3]
```diff
@@ -3,13 +3,15 @@
...
 
+import com.azure.android.communication.common.CommunicationIdentifier
 import com.azure.android.communication.common.CommunicationTokenCredential
 import com.azure.android.communication.ui.calling.DiagnosticConfig
 import java.util.UUID
 
 internal enum class CallType {
     GROUP_CALL,
     TEAMS_MEETING,
+    ONE_ON_ONE_CALL
```
Comment (iaulakh) [inline azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/configuration/CallConfiguration.kt:14]: Please change to ONE_TO_N_CALL

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/CallComposite.java [hunk 4]
```diff
@@ -335,22 +338,33 @@ private void launchComposite(final Context context,
...
+            } else {
+                callType = CallType.TEAMS_MEETING;
+                meetingLink = ((CallCompositeTeamsMeetingLinkLocator) locator).getMeetingLink();
+            }
+        }
+        else {
+            callType = CallType.ONE_ON_ONE_CALL;
+            participants = new ArrayList<>();
+            for (String participantId : remoteOptions.getStartCallOptions().getParticipants()) {
+                participants.add(new CommunicationUserIdentifier(participantId));
```
Comment (iaulakh) [inline azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/CallComposite.java:357]: Please change to CommunicationIdentifier.fromRawId(participantId)

Comment (iaulakh) [reply azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/CallComposite.java:357]: We do not need to convert here. We can use list of string until reaches to service layer

File: azure-communication-ui/demo-app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -10,7 +10,7 @@
     <string name="identity_hint">Identity</string>
     <string name="group_call_label">Group call</string>
     <string name="teams_meeting_label">Teams meeting</string>
-    <string name="group_call_id_or_teams_meeting_hint">Group ID or Meeting link</string>
+    <string name="group_call_id_or_teams_meeting_hint">Group ID or Meeting link or Participant Id</string>
```
Comment (iaulakh) [inline azure-communication-ui/demo-app/src/main/res/values/strings.xml:13]: Id's comma separated

File: azure-communication-ui/demo-app/build.gradle [hunk 6]
```diff
@@ -31,6 +31,7 @@ android {
         buildConfigField 'String', 'THREAD_ID', properties.getProperty('THREAD_ID', '""')
         buildConfigField 'String', 'END_POINT_URL', properties.getProperty('END_POINT_URL', '""')
         buildConfigField "String", 'AAD_TOKEN', properties.getProperty('AAD_TOKEN', '""')
+        buildConfigField 'String', 'PARTICIPANT_MRI', properties.getProperty('PARTICIPANT_MRI', '""')
```
Comment (iaulakh) [inline azure-communication-ui/demo-app/build.gradle:34]: can we plase add support for multiple id's. separated by comma

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/models/CallCompositeRemoteOptions.java [hunk 7]
```diff
@@ -91,4 +102,13 @@ public String getDisplayName() {
     public CallCompositeJoinLocator getLocator() {
         return locator;
     }
+
+    /**
+     * Get call options.
```
[the remark this answers is not in the document]
Comment (pratimag1289) [reply azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/models/CallCompositeRemoteOptions.java]: Done

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/models/CallCompositeStartCallOptions.java [hunk 8]
```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.communication.ui.calling.models;
+
+import java.util.List;
+
+public class CallCompositeStartCallOptions {
```
[the remark this answers is not in the document]
Comment (pratimag1289) [reply azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/models/CallCompositeStartCallOptions.java:13]: Done.

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/models/CallCompositeStartCallOptions.java [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+    private final List<String> participants;
+
+    public CallCompositeStartCallOptions(final List<String> participants) {
```
[the remark this answers is not in the document]
Comment (pratimag1289) [reply azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/models/CallCompositeStartCallOptions.java:20]: @pavelprystinka, this is Java code hence making it private with getter would make it a read-only property.

**Review by iaulakh — APPROVED** [review]

**Pull request merged**